catalyst-gateway/bin/src/rbac_cache/cache_manager.rs

```diff
@@ -35,18 +48,29 @@ impl RbacCacheManager {
     #[allow(clippy::result_large_err)]
     pub fn add(&self, registration: Cip509, is_persistent: bool) -> AddResult {
         if is_persistent {
-            self.persistent.add(registration.clone())?;
+            self.persistent.add(registration.clone())
+        } else {
+            self.volatile.add(registration)
```

Comment:
The previous implementation was intentional: the `RegistrationChain` object must always contain the full chain, so the volatile cache contains both persistent and volatile parts.